packages/core/src/new-api/internal/reconciliation/reconcilier.ts

```diff
@@ -0,0 +1,115 @@
+import { ExecutionState, ExecutionStateMap } from "../../types/execution-state";
+import { Future, IgnitionModule, ModuleParameters } from "../../types/module";
+import { AdjacencyList } from "../utils/adjacency-list";
+import { AdjacencyListConverter } from "../utils/adjacency-list-converter";
+
+import { reconcileCurrentAndPreviousTypeMatch } from "./reconcileCurrentAndPreviousTypeMatch";
+import { reconcileDependencyRules } from "./reconcileDependencyRules";
+import { reconcileFutureSpecificReconciliations } from "./reconcileFutureSpecificReconciliations";
+import {
+  ReconciliationCheck,
+  ReconciliationContext,
+  ReconciliationFailure,
+  ReconciliationFutureResult,
+  ReconciliationFutureResultFailure,
+  ReconciliationResult,
+} from "./types";
+
+export class Reconcilier {
+  public static reconcile(
+    module: IgnitionModule,
```

Comment:
Maybe worth making it explicit if this includes the futures of submodules

Reply:
I'm pretty sure it does, but while reviewing this file I found myself having to double check it by reading the code.

Reply:
Agreed, would changing the name of the function to `reconcileModuleAndSubmodules` clarify, or is there a different approach (e.g. a dreaded comment)?